Automattic/wp-calypso
# Create NavigationHeader for DomainHeader

Fixes https://github.com/Automattic/dotcom-forge/issues/3353

## Proposed Changes

Context: p9Jlb4-8ec

> [!NOTE]  
> Nothing functional has been changed so this review can be focused on visual aspects.


* Changes to the <DomainHeader> component to align with the new consolidated Header design
* That is a total of 10 instances
(checklist to help with testing)
- [x] 1. Connect another domain: `/domains/mapping/:siteSlug/setup/:connectToDomain` File: `client/components/domains/connect-domain-step/index.jsx`
- [x] 2. DNS > Add a DNS Record: `/domains/manage/all/:domain/add-dns-record/:siteSlug` File: `client/my-sites/domains/domain-management/dns/add-dns-record.jsx`
- [x] 3. DNS: `/domains/manage/all/:domain/dns/:siteSlug` File: `client/my-sites/domains/domain-management/dns/dns-records.jsx`
- [x] 4. Edit contact information: `/domains/manage/all/:domain/edit-contact-info/:siteSlug` File: `client/my-sites/domains/domain-management/edit-contact-info-page/index.tsx`
- [x] 5. Domain management: `/domains/manage` File: `client/my-sites/domains/domain-management/list/all-domains.jsx`
- [x] 6. Domain management (site): `/domains/manage/:siteSlug` File: `client/my-sites/domains/domain-management/list/site-domains.jsx`
- [x] 7. Domain Settings: `/domains/manage/all/:domain/edit/:siteSlug` File: `client/my-sites/domains/domain-management/settings/index.tsx`
- [x] 8. Transfer: `/domains/manage/all/:domain/transfer/:siteSlug` File: `client/my-sites/domains/domain-management/transfer/transfer-page/index.tsx`
- [x] 9. Transfer > Transfer to another WP.com site: `/domains/manage/all/:domain/transfer/other-site/:siteSlug` File: `client/my-sites/domains/domain-management/transfer/transfer-to-other-site/transfer-domain-to-other-site.tsx`
- [x] 10. Transfer > Transfer to another user: `/domains/manage/all/:siteSlug/transfer/other-user/:siteSlug` File: `client/my-sites/domains/domain-management/transfer/transfer-to-other-user/transfer-domain-to-other-user.jsx`

Here's the Header's intended behavior (domain data redacted, only showing the Header)

### Before:

https://github.com/Automattic/wp-calypso/assets/6586048/7a8c32c8-f792-4100-ac01-55b584745018

### After:

https://github.com/Automattic/wp-calypso/assets/6586048/960c69c0-b8b1-4171-829d-ce0726d76cb2

## Testing Instructions



* Please check the checklist above for all the pages that need testing
* Test for all screen sizes

| Page | Screenshots |
| :-------------: | :-------------: |
| Connect another domain | <img src="https://github.com/Automattic/wp-calypso/assets/6586048/ab4768e8-2919-4e88-a7e9-756edca5761e">  |
| DNS > Add a DNS Record | <img src="https://github.com/Automattic/wp-calypso/assets/6586048/a484d815-7601-4fb4-9581-661fed144c1f"> |
| DNS | <img src="https://github.com/Automattic/wp-calypso/assets/6586048/86f88571-7fa0-43bb-bf93-0de12e17e4ad"> |
| Edit contact information | <img src="https://github.com/Automattic/wp-calypso/assets/6586048/6937be7f-0e93-41cb-8cbb-06d6109580b1"> |
| Domain management | <img src="https://github.com/Automattic/wp-calypso/assets/6586048/7680d697-625f-4d07-8c18-3bc70ddc644d"> |
| Domain management (site) | <img src="https://github.com/Automattic/wp-calypso/assets/6586048/5961dcff-030f-49f2-b279-0dbb6dccddc6"> |
| Domain Settings | <img src="https://github.com/Automattic/wp-calypso/assets/6586048/faa2a7e6-977d-4b59-acda-818ab9a75ff7"> |
| Transfer | <img src="https://github.com/Automattic/wp-calypso/assets/6586048/bf5e64aa-e5bb-4a89-b244-554cb5607b7e"> |
| Transfer > Transfer to another WP.com site | <img src="https://github.com/Automattic/wp-calypso/assets/6586048/0da10f71-5e75-4475-8328-c0d91d5b0986"> |
| Transfer > Transfer to another user | <img src="https://github.com/Automattic/wp-calypso/assets/6586048/bb4898a2-9db6-48c9-8f1a-cedf673e1608"> |

## Pre-merge Checklist
